docs/general/ad-filtering/create-own-filters.md

```diff
@@ -1737,6 +1740,210 @@ As a response to blocked request AdGuard returns a short video placeholder.
+
+Specified domains through `$domain` or specified applications through `$app` add `100 + 100 / N`, where `N` is the number of modifier values for example:
+`||example.com^$domain=example.com|example.org|example.net` will add `100 + 100 / 3 = 134.3 = 135` or
+`||example.com^$app=org.example.app1|org.example.app2` will add `100 + 100 / 2 = 151` or
+`||example.com^$domain=example.com,app=org.example.app1|org.example.app2` will add `100 + 100/1` ($domain part) and `100 + 100/2` ($app part) - will add `350` in total.
+
+`$domain` with regexps or tld values adds only 1 for the sake of simplicity.
+
+#### 4. Specific exceptions: {#priority-category-4}
+
+[//]: # (Please keep them sorted)
+
+* [`$content`](#content-modifier),
+* [`$elemhide`](#elemhide-modifier),
+* [`$extension`](#extension-modifier),
+* [`$jsinject`](#jsinject-modifier),
+* [`$specifichide`](#specifichide-modifier),
+* [`$urlblock`](#urlblock-modifier),
+* [`$genericblock`](#genericblock-modifier),
+* [`$generichide`](#generichide-modifier);
+
+Each of which adds `10^3` to the priority.
+
+As well as exception with [`$document modifier`](#document-modifier): because it's an alias for `$elemhide,content,jsinject,urlblock,extension`.
+It will add `10^3` for each modifier from [the top list](#priority-category-4), `10^3 * 5` in total.
+
+In addition, each of these exceptions implicitly adds the two allowed content-type modifiers `$document,subdocument`.
+
+#### 5. Allowlist rule {#priority-category-5}
+
+Modifier `@@` adds `10^4` to rule priority.
+
+#### 6. $redirect rules {#priority-category-6}
+
+[//]: # (Please keep them sorted)
+
+* [`$redirect`](#redirect-modifier),
+* [`$redirect-rule`](#redirect-rule-modifier).
+
+Each of which adds `10^5` to rule priority.
+
+#### 7. important rules {#priority-category-7}
+
+Modifier [`$important`](#important-modifier) adds `10^6` to rule priority.
+
+#### Rules for which there is no priority count {#priority-category-extra}
+
+[Other modifiers](#advanced-capabilities), which do not override a lock but perform additional post- or pre-processing of requests, have no priority.
+
+> **Note**
+>
+> The [`$replace`](#replace-modifier) modifier takes precedence over all blocking rules of categories 1-3, as well as rules of exceptions from categories 3-5, **except** [`$content`](#content-modifier), because an exception with the `$content` modifier overrides all `$replace` rules.
+
+#### Explanation for complicated cases:
+
+1. Simultaneous blocking and `$removeparam`.
+For example `*&p1=param` and `*$removeparam=p1`.
+The blocking rule will be applied first, because otherwise, after removing the parameter, the request will not be blocked.
+
+2. `$header` and `$removeheader`.
+Two rules:
+`||example.com^$header=header-name` - blocking.
+`||example.com^$removeheader=header-name` - remove header.
+The blocking rule will be applied first, because `$header` will add 1 to the priority calculation.
```

Comment:
Have you decided to leave it here? The `removeparam` case explanation is now duplicated in the modifier's section, but the `header` case is not. 🤔

Reply:
yes, because if I duplicate it, I have to duplicate it twice, and we will have three identical partitions: here and in `$header` and `$remove-header`

Reply:
I'm still not sure it's a good place to leave it here. Those cases have nothing common with the priority. Maybe, it would be better to extract them into a separate chapter?

Reply:
@sxgunchenko can you provide sample how you see it? under "> note" section or how?

Reply:
I see at least 2 options:

1) Introduce a chapter after https://github.com/AdguardTeam/KnowledgeBase/blob/master/docs/general/ad-filtering/create-own-filters.md?plain=1#LL796C1-L796C26, like:
```
### Tricky cases
1. Simultaneous blocking and `$removeparam`...
2. `$header` and `$removeheader`...
```

2) Add notes to `$header` and `$remove-header` section, just like you've done with `$removeparam`.

I would prefer the second one because the first one still requires links in the correspondent modifier section to the chapter. In the second variant it's not necessary to duplicate the case description, one of them could contain the full description and another - just a link.

Reply:
@sxgunchenko done, thank you

Reply:
it's still here 🤔

Reply:
deleted at all